### PR TITLE
[IMP] html_editor: cache the metadata for link previews

### PR DESCRIPTION
**Pull request by Jinjiu96**
Before this commit: An RPC is made each time to retrieve the metadata
for link previews.

After this commit: The retrieved metadata is cached in a dictionary, so
only the first link preview requires an RPC.

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/html_editor/static/src/main/link/link_popover.js [hunk 1]
```diff
@@ -197,75 +200,45 @@ export class LinkPopover extends Component {
...
+                externalMetadata?.og_image &&
+                this.state.label &&
+                this.state.urlTitle === this.state.url
+            ) {
                 this.state.urlTitle = this.state.label;
             }
         } else {
-            await this.keepLastPromise
+            const html_parser = new window.DOMParser();
+            // Get
```
Comment (cammarosano) [inline addons/html_editor/static/src/main/link/link_popover.js:226]: I think you forgot to delete this comment line ("Get")

Comment (Jinjiu96) [reply addons/html_editor/static/src/main/link/link_popover.js:226]: Nope, I actually put it there to align with  the `//set` below :rofl:

Comment (FrancoisGe) [reply addons/html_editor/static/src/main/link/link_popover.js:226]: What do you mean by "GET" ? It s not really clear ?

Comment (Jinjiu96) [reply addons/html_editor/static/src/main/link/link_popover.js]: Just updated the comments, I used `set` and `get` as "titles" but can be a bit redundant :grimacing:

File: addons/html_editor/static/src/main/link/link_popover.js [hunk 2]
```diff
@@ -6,6 +6,9 @@ import { cleanZWChars, deduceURLfromText } from "./utils";
 import { KeepLast } from "@web/core/utils/concurrency";
 import { rpc } from "@web/core/network/rpc";
 
+window.externalLinkCache = {};
```
Comment (fdardenne) [inline addons/html_editor/static/src/main/link/link_popover.js:9]: Put them in const that you can export

**Review by fdardenne — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**